### PR TITLE
Fix iobuf inference

### PR DESCRIPTION
**Pull request by mkurc-ant**
This PR fixes IOBUF inference during synthesis for ROI and ROI-less targets. The USE_ROI is passed to the TCL synthesis script where it is used to determine whether to infer IOBs or not.

### CODE REVIEW

File: xc7/yosys/synth.tcl [hunk 1]
```diff
@@ -1,22 +1,18 @@
+# Import the commands from the plugins to the tcl interpreter
 yosys -import
 
 plugin -i xdc
 plugin -i fasm
-#Import the commands from the plugins to the tcl interpreter
-yosys -import
```
Comment (litghost) [inline xc7/yosys/synth.tcl]: Isn't this required?

Comment (mkurc-ant) [reply xc7/yosys/synth.tcl]: But it's done in the first line of the script.

**Comment by mkurc-ant** [comment]
I haven't checked my changes thoroughly. I'm waiting what the CI says.

File: xc7/yosys/synth.tcl (hunk 1 above)
Comment (litghost) [reply xc7/yosys/synth.tcl]: If you read the comment, the `yosys -import` imports new commands that come from the XDC and FASM plugins.

Comment (mkurc-ant) [reply xc7/yosys/synth.tcl]: Ahh..., you're right.

**Comment by litghost** [comment]
CI is red.  Worth noting that io buf inference isn't really ready for general use, see: https://github.com/YosysHQ/yosys/issues/1720

**Comment by mkurc-ant** [comment]
I've read that. But unfortunately we have to infer IO bufs as many designs like eg. the Litex genrated ones do not instance them. And if there are no IO bufs then the XDC plugin has nothing to attach IOSTANDARD settings to.

**Comment by litghost** [comment]
> I've read that. But unfortunately we have to infer IO bufs as many designs like eg. the Litex genrated ones do not instance them. And if there are no IO bufs then the XDC plugin has nothing to attach IOSTANDARD settings to.

I agree we need to have working IO buf inference, I'm just worried that the broken `iopadmap` implementation will cause many failures.  We may need to patch yosys before we enable inference in symbiflow-arch-defs.

**Comment by litghost** [comment]
Superceded by #1349

**Pull request closed without merging**